log/src/lib.rs

```diff
@@ -0,0 +1,119 @@
+use lv2_core::feature::*;
+//use std::ffi::CString;
+use std::fmt;
+use std::os::raw::*; //get all common c_type
+use urid::*;
+
+pub struct EntryClass;
+unsafe impl UriBound for EntryClass {
+    const URI: &'static [u8] = lv2_sys::LV2_LOG__Entry;
+}
+
+pub struct ErrorClass;
+unsafe impl UriBound for ErrorClass {
+    const URI: &'static [u8] = lv2_sys::LV2_LOG__Error;
+}
+
+pub struct NoteClass;
+unsafe impl UriBound for NoteClass {
+    const URI: &'static [u8] = lv2_sys::LV2_LOG__Note;
+}
+
+pub struct TraceClass;
+unsafe impl UriBound for TraceClass {
+    const URI: &'static [u8] = lv2_sys::LV2_LOG__Trace;
+}
+
+pub struct WarningClass;
+unsafe impl UriBound for WarningClass {
+    const URI: &'static [u8] = lv2_sys::LV2_LOG__Warning;
+}
+
+/// Marker for URID representing the nature of a log message
+// Note : it's may be better to have a URID trait to define a common interface
+pub unsafe trait EntryType {
+    fn get(self) -> u32;
+}
+
+unsafe impl EntryType for URID<ErrorClass> {
```

Comment:
I believe it's better to implement `EntryType` on each of the entry type structs (`TraceClass`, `ErrorClass`) and then receive an `URID<impl EntryType>` rather than an `impl EntryType`.

Not only this makes you have to write less code (you don't have to reimplement `get()` every time), but also the method still receives a full URID with all of its capabilities, and if anything it's more in line with what we do everywhere else that uses URIDs within the `lv2` crate (features, extensions, atom types, etc.) :slightly_smiling_face:

Reply:
thanks, i missed this trick.